packages/blade/src/components/Typography/BaseText/BaseText.d.ts

```diff
@@ -0,0 +1,32 @@
+import type { ReactNode } from 'react';
+import type { ColorContrast } from '../../../tokens/theme/theme';
+import type { Theme } from '../../BladeProvider';
+
+type DotNotationStringToken<TokenType> = {
+  [K in keyof TokenType]: `${K & string}.${TokenType[K] extends Record<
+    string,
+    ColorContrast | string
+  >
+    ? Extract<keyof TokenType[K], number | string>
+    : DotNotationStringToken<TokenType[K]>}`;
+}[keyof TokenType];
+
+type FeedbackColors = `feedback.text.${DotNotationStringToken<
```

Comment:
Not very fond of not having type checking in a file which has complex types. 

Either we can change this file to `.ts` or add 

```ts
  "compilerOptions": {
    "skipLibCheck": false
  },
```

in tsconfig to enable typechecking of d.ts files

Reply:
but that will enable checks for tall the internal libs as well right?